packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/synthesize-resources.ts

```diff
@@ -0,0 +1,340 @@
+import { ServiceQuestionsResult } from '../service-walkthrough-types';
+import path from 'path';
+import fs, { existsSync, copySync } from 'fs-extra';
+import uuid from 'uuid';
+import { cfnTemplateRoot, privateKeys, adminAuthAssetRoot, triggerRoot, ENV_SPECIFIC_PARAMS } from '../constants';
+import { ServiceName as FunctionServiceName } from 'amplify-category-function';
+
+const category = 'auth';
+
+/**
+ * Factory function that returns a function that synthesizes all resources based on a ServiceQuestionsResult request.
+ * The function returns the request unchanged to enable .then() chaining
+ * @param context The amplify context
+ * @param cfnFilename The template cfnFilename
+ * @param provider The cloud provider name
+ */
+export const getResourceSynthesizer = (context: any, cfnFilename: string, provider: string) => async (
+  request: Readonly<ServiceQuestionsResult>,
+) => {
+  await lambdaTriggers(request, context, null);
+  await createUserPoolGroups(context, request.resourceName, request.userPoolGroupList);
+  await addAdminAuth(context, request.resourceName!, 'add', request.adminQueryGroup);
+  await copyCfnTemplate(context, category, request, cfnFilename);
+  saveResourceParameters(context, provider, category, request.resourceName!, request, ENV_SPECIFIC_PARAMS);
+  await copyS3Assets(context, request);
+  return request;
+};
+
+/**
+ * The functions below should not be exported, but they are for now because the update flow still uses them individually
+ * Once the update flow is also refactored, they will be internal to this file
+ *
+ * They are refactored with minimal modification from awscloudformation/indes.js
+ */
+
+export const lambdaTriggers = async (coreAnswers: any, context: any, previouslySaved: any) => {
+  const { handleTriggers } = require('./trigger-flow-auth-helper');
+  let triggerKeyValues = {};
+
+  if (coreAnswers.triggers) {
+    triggerKeyValues = await handleTriggers(context, coreAnswers, previouslySaved);
+    coreAnswers.triggers = triggerKeyValues ? JSON.stringify(triggerKeyValues) : '{}';
+
+    if (triggerKeyValues) {
+      coreAnswers.parentStack = { Ref: 'AWS::StackId' };
+    }
+
+    // determine permissions needed for each trigger module
+    coreAnswers.permissions = await context.amplify.getTriggerPermissions(context, coreAnswers.triggers, 'auth', coreAnswers.resourceName);
+  } else if (previouslySaved) {
+    const targetDir = context.amplify.pathManager.getBackendDirPath();
+    Object.keys(previouslySaved).forEach(p => {
+      delete coreAnswers[p];
+    });
+    await context.amplify.deleteAllTriggers(previouslySaved, coreAnswers.resourceName, targetDir, context);
+  }
+  // remove unused coreAnswers.triggers key
+  if (coreAnswers.triggers && coreAnswers.triggers === '[]') {
```

Comment:
Is the string '[]` check because `triggers` is a stringified JSON?

Reply:
This is all copied over from existing code, but most likely because I see some JSON.parse(answers.triggers) in other places